src/rounded_rect.rs

```diff
@@ -1,57 +1,73 @@
 //! A rectangle with rounded corners.
 
-use crate::{arc::ArcAppendIter, Arc, PathEl, Point, Rect, Shape, Size, Vec2};
-use std::f64::consts::{FRAC_PI_2, PI};
+use crate::{arc::ArcAppendIter, Arc, PathEl, Point, Rect, RoundedRectRadii, Shape, Size, Vec2};
+use std::f64::consts::{FRAC_PI_2, FRAC_PI_4};
 
 /// A rectangle with equally rounded corners.
 ///
 /// By construction the rounded rectangle will have
-/// non-negative dimensions and radius clamped to half size of the rect.
+/// non-negative dimensions and radii clamped to half size of the rect.
 ///
 /// The easiest way to create a `RoundedRect` is often to create a [`Rect`],
 /// and then call [`to_rounded_rect`].
 ///
+/// Corner radius values can be set by directly passing a [`RoundedRectRadii`]
+/// to [`RoundedRect::new`] or [`Rect::to_rounded_rect`]. [`RoundedRectRadii`]
+/// also provides type conversion from `f64` and `(f64, f64, f64, f64)`:
+///
+/// ```
+/// use kurbo::{RoundedRect, RoundedRectRadii};
+///
+/// // Create a rounded rectangle with a single radius for all corners:
+/// RoundedRect::new(0.0, 0.0, 10.0, 10.0, 5.0.into());
+///
+/// // Or, specify different radii for each corner, clockwise from the top-left:
+/// RoundedRect::new(0.0, 0.0, 10.0, 10.0, (1.0, 2.0, 3.0, 4.0).into());
+/// ```
+///
 /// [`to_rounded_rect`]: Rect::to_rounded_rect
 #[derive(Clone, Copy, Default, Debug, PartialEq)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub struct RoundedRect {
     /// Coordinates of the rectangle.
     rect: Rect,
     /// Radius of all four corners.
-    radius: f64,
+    radii: RoundedRectRadii,
 }
 
 impl RoundedRect {
     /// A new rectangle from minimum and maximum coordinates.
     ///
-    /// The result will have non-negative width, height and radius.
+    /// The result will have non-negative width, height and radii.
     #[inline]
-    pub fn new(x0: f64, y0: f64, x1: f64, y1: f64, radius: f64) -> RoundedRect {
-        RoundedRect::from_rect(Rect::new(x0, y0, x1, y1), radius)
+    pub fn new(x0: f64, y0: f64, x1: f64, y1: f64, radii: RoundedRectRadii) -> RoundedRect {
```

Comment:
A possibility here is to have `impl Into<RoundedRectRadii>` here, which would be more ergonomic for allowing a simple scalar radius.

Reply:
Ah of course, my lack of Rust knowledge is showing here. Thanks for the suggestion!